### PR TITLE
fetch schema job impl

### PR DESCRIPTION
**Pull request by shz117**
- fetch schema job impl
 - ares controller client impl
 - updateTable impl in diskstore

### CODE REVIEW

File: metastore/schema_fetch.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	"github.com/uber/aresdb/utils"
+	"reflect"
+	"time"
+)
+
+// SchemaFetchJob is a job that periodically pings ares-controller and updates table schemas if applicable
+type SchemaFetchJob struct {
+	clusterName      string
+	hash             string
+	interval         int
```
Comment (jshencode) [inline metastore/schema_fetch.go]: maybe rename to intervalInSeconds will be more clear

File: clients/controller.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	if err != nil {
+		return
+	}
+	var resp *http.Response
+	resp, err = c.c.Do(req)
+	if err != nil {
+		return
+	}
+	if resp.StatusCode != http.StatusOK {
+		err = utils.StackError(fmt.Errorf("controller client error fetching schema"), fmt.Sprintf("code %d", resp.StatusCode))
```
Comment (jshencode) [inline clients/controller.go]: controller error fetching schema could be just in the error message

File: cmd/cmd.go [hunk 3]
```diff
@@ -152,6 +153,17 @@ func StartService(cfg common.AresServerConfig, logger bark.Logger, queryLogger b
 	batchStatsReporter := memstore.NewBatchStatsReporter(5*60, memStore, metaStore)
 	go batchStatsReporter.Run()
 
+	if cfg.ClusterName != "" {
```
Comment (jshencode) [inline cmd/cmd.go]: also, when we are in cluster mode, schema handler should be open only on debug port

File: api/schema_handler.go [hunk 4]
```diff
@@ -128,6 +131,10 @@ func (handler *SchemaHandler) GetTable(w http.ResponseWriter, r *http.Request) {
 //    default: errorResponse
 //        200: noContentResponse
 func (handler *SchemaHandler) AddTable(w http.ResponseWriter, r *http.Request) {
+	if handler.isClusterMode {
```
Comment (jshencode) [inline api/schema_handler.go]: we can modify Register to just register the read/Get calls

Comment (shz117) [reply api/schema_handler.go]: i'm afraid that would confuse user with 404 error. 405 (not allowed) may make more sense?

File: cmd/cmd.go (hunk 3 above)
Comment (shz117) [reply cmd/cmd.go]: does it make sense to still open read methods?

Comment (shz117) [reply cmd/cmd.go]: I was considering styx/subscriber will still need to read schema from instance during migration

File: clients/controller.go (hunk 2 above)
Comment (shz117) [reply clients/controller.go]: what do you mean?

Comment (jshencode) [reply clients/controller.go]: "controller client error" could be in the message

File: api/schema_handler.go (hunk 4 above)
Comment (jshencode) [reply api/schema_handler.go]: hmm... i think we can use router.Methods() to control the methods

**Pull request merged**